source/CIMenuScene.cpp

```diff
@@ -52,185 +55,104 @@ bool MenuScene::init(const std::shared_ptr<AssetManager>& assets) {
                 break;
-            case MenuStatus::GameLobby:
-                _status = MenuStatus::LobbyToMain;
+            case MenuState::GameLobby:
+                _state = MenuState::LobbyToMain;
                 break;
-            case MenuStatus::Tutorial:
-                _status = MenuStatus::TutorialToMain;
+            case MenuState::Tutorial:
+                _state = MenuState::TutorialToMain;
                 break;
             default:
                 break;
             }
         }
         });
-        
-    /** Handle non-button input listeners */
-    /** Volume slider listeners */
-    _volume = _volumeSlider->getValue(); // update volume using Volume Slider in Settings 
-    _volumeSlider->addListener([=](const std::string& name, float value) {
-        if (value != _volume) {
-            // TODO: Update global game volume
-            _volume = value;
-        }
-        });
-    /** Player name input listeners */
-    _pnameInput->addTypeListener([=](const std::string& name, const std::string& value) {
-        // Handle size limit on settings name textfield
-        if (value.length() > 12) {
-            _pnameInput->setText(value.substr(0, 12));
-        }
-        });
-    _pnameInput->addExitListener([=](const std::string& name, const std::string& value) {
-        // Set player name input
-        _playerName = value;
-        CULog("Player Name set to %s", value.c_str());
-        });
-    /** Join game room id input listeners */
-    _roomIdInput->addTypeListener([=](const std::string& name, const std::string& value) {
-        // Handle room id textfield with size limit
-        if (value.length() > 5) {
-            _roomIdInput->setText(value.substr(value.length() - 5, 5));
-        }
-        });
-    _roomIdInput->addExitListener([=](const std::string& name, const std::string& value) {
-        // Set game room id input 
-        _joinGame = value;
-        CULog("Room id set to %s", value.c_str());
-        });
 
-    /** Set both toggle buttons. */
-    _musicBtn->setToggle(true);
-    _parallaxBtn->setToggle(true);
+    // TODO: integrate network manager to game lobby
+
+    _playerName = playerName;
+    _volume = volume;
+    _musicOn = musicOn;
+    _parallaxOn = parallaxOn;
+
+    _otherNames = vector<string>{ "N/A", "N/A", "N/A", "N/A" };
+    _joinGame = "";
+
+    _state = MenuState::LoadToMain;
 
-    _status = MenuStatus::MainMenu;
     Application::get()->setClearColor(Color4(192, 192, 192, 255));
     addChildWithName(layer, "menuScene");
+
+    _mainmenu = MainMenu::alloc(_assets);
+    _mainmenu->setDisplay(false);
+
+    _tutorial = TutorialMenu::alloc(_assets);
+    _tutorial->setDisplay(false);
+
+    _settings = SettingsMenu::alloc(_assets);
+    _settings->setDisplay(false);
+
+    _join = JoinMenu::alloc(_assets);
+    _join->setDisplay(false);
+
+    _lobby = LobbyMenu::alloc(_assets);
+    _lobby->setDisplay(false);
+
+    addChild(_mainmenu->getLayer(), 3);
```

Comment:
Is there a reason you put 3 here? I don't think giving all the children the same z-value does anything